src/cobalt.rs

```diff
@@ -231,73 +232,38 @@ fn create_rss(path: &str, dest: &Path, config: &Config, documents: &[Document])
-        let mut content_splits = content2.split("---");
-
-        // above the split are the attributes
-        let attribute_string = content_splits.next().unwrap_or("");
-
-        // everything below the split becomes the new content
-        content = content_splits.next().unwrap_or("").to_owned();
-
-        let yaml_result = try!(YamlLoader::load_from_str(attribute_string));
-
-        let yaml_attributes = try!(yaml_result[0]
-                                       .as_hash()
-                                       .ok_or(format!("Incorrect front matter format in {:?}",
-                                                      path)));
-
-        for (key, value) in yaml_attributes {
-            // TODO is unwrap_or the best way to handle this?
-            attributes.insert(key.as_str().unwrap_or("").to_owned(),
-                              value.as_str().unwrap_or("").to_owned());
+/// A slightly less efficient implementation of fs::create_dir_all
+/// that eliminates the race condition problems of the original
+fn create_dir_all(path: &Path) -> io::Result<()> {
+    let mut new_path = PathBuf::new();
+    for component in path {
+        new_path.push(component);
+        match fs::create_dir(&new_path) {
+            Ok(_) => {}
+            Err(ref e) if e.kind() == ErrorKind::AlreadyExists => {}
+            Err(e) => return Err(e),
         }
     }
+    Ok(())
+}
 
-    let date = attributes.get("date")
-                         .and_then(|d| DateTime::parse_from_str(d, "%d %B %Y %H:%M:%S %z").ok());
-
-    let path_str = try!(path.to_str()
-                            .ok_or(format!("Cannot convert pathname {:?} to UTF-8", path)));
-
-    let source_str = try!(source.to_str()
-                                .ok_or(format!("Cannot convert pathname {:?} to UTF-8", source)));
-
-    let new_path = try!(path_str.split(source_str)
-                                .last()
-                                .ok_or(format!("Empty path")));
-
-    // construct path
-    let mut path_buf = PathBuf::from(new_path);
-    path_buf.set_extension("html");
-
-    let path_str = try!(path_buf.to_str()
-                                .ok_or(format!("Cannot convert pathname {:?} to UTF-8", path_str)));
+fn create_document_file<T: AsRef<Path>>(content: String, path: T, dest: &Path) -> Result<()> {
+    // construct target path
+    let mut file_path_buf = PathBuf::new();
+    file_path_buf.push(dest);
+    file_path_buf.push(path);
 
-    let markdown = path.extension().unwrap_or(OsStr::new("")) == OsStr::new("md");
+    let file_path = file_path_buf.as_path();
 
-    let name = try!(path.file_stem()
-                        .and_then(|stem| stem.to_str())
-                        .ok_or(format!("Invalid UTF-8 in file stem for {:?}", path)));
+    // create target directories if any exist
+    if let Some(parent) = file_path.parent() {
+        try!(create_dir_all(parent).map_err(|e| format!("Could not create {:?}: {}", parent, e)));
```

Comment:
It does that, it appends the io message to the other text. This way the user gets some info on what file actually failed to create.